File: IKeep.Lib/Models/ElementType.cs
using IKeep.Lib.Core;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace IKeep.Lib.Models
{
    public class ElementType : Entity
    {
        public string Ref { get; set; }
        public string Name { get; set; }

        [JsonIgnore]
        public virtual ICollection<GenericElement> GenericElements { get; set; }
        public List<Guid> GenericElementsIds
        {
            get
            {
                return GenericElements == null ? new List<Guid>() : GenericElements.Select(x => x.Id).ToList();
            }
        }
    }
}

File: IKeep.Web/ClientApp/src/models/ElementType.js
import Entity from './Entity';

class ElementType extends Entity
{
    constructor(json)
    {
        super(json)
        if(json)
        {
            this.Ref = json.ref;
            this.Name = json.name;
            this.GenericElementsIds = json.genericElementsIds;
        }
        else
        {
            this.Ref = "";
            this.Name = "";
            this.GenericElementsIds = [];
        } 
    }
}
export default ElementType;